git_downloader.py

```diff
@@ -16,33 +17,51 @@ def main(args, loglevel):
     logging.basicConfig(format="%(levelname)s: %(message)s", level=loglevel)
 
     socket.setdefaulttimeout(args.timeout)
-    
-    g = Github()
-    with open(args.repo_file, 'r') as f:
-        file_counter = 0
-        for line in f.readlines():
-            logging.info('Fetching repository: %s' % line)
-            try:
-                repo_str = line.rstrip().split('github.com/')[-1]
-                repo = g.get_repo(repo_str)
-                tree = repo.get_git_tree('master', recursive=True)
-                files_to_download = []
-                for file in tree.tree:
-                    if fnmatch.fnmatch(file.path, args.wildcard):
-                        files_to_download.append('https://github.com/%s/raw/master/%s' % (repo_str, file.path))
-                for file in files_to_download:
-                    logging.info('Downloading %s' % file)
-                    file_counter += 1
-                    filename = posixpath.basename(urlparse.urlsplit(file).path)
-                    output_path = os.path.join(args.output_dir, filename)
-                    if os.path.exists(output_path):
-                        output_path += "-" + str(file_counter)
-                    try:
-                        urlretrieve(file, output_path)
-                    except Exception as e:
-                        logging.exception('Error downloading %s.' % file)
-            except Exception as e:
-                 logging.exception('Error fetching repository %s.' % line)
+
+    g = github.Github()
+
+    if args.repo_file:
+        with open(args.repo_file, 'r') as f:
+            repo_gen = f
+            download_files(args, g, repo_gen)
+    else:
+        def repo_gen():
+            last_page = args.crawl_since
+            git_repo_gen = g.get_repos(since=args.crawl_since)
+            for repo in git_repo_gen:
+                current_page = git_repo_gen.__nextUrl
```

Comment:
If you don't find my answer satisfying, please followup in PR #6 with how you'd like to proceed 

Should I remove the print?
Should I avoid using `__netxtUrl`?

(keep in mind this will prevent resuming a stopped execution easily)